public/dorequest.php

```diff
@@ -320,8 +320,9 @@ function DoRequestError($errorMsg)
 
     case "lbinfo":
         $lbID = requestInput('i', 0, 'integer');
+        $nearby = 1;
```

Comment:
`$nearby` should only be true if `$user` is provided. There's still a desire to request `$count` items starting at `$offset` through this API.

Reply:
Thanks for clarifying that I'll look into this, I'm still a bit confused on the dll side of things I don't see where it passes the userjinto the call but am assuming it does somewhere

Reply:
the dll passes `u` and `t` to every `dorequest` API call. I didn't realize it's not required for all of them. in this case, it's a happy accident that changes the DLL behavior without having to change the DLL code. as you suggested, I'll probably change it anyway to fetch one more record for visual balance.

the `$count/$offset` usage would be new for scrolling the list or possibly for fetching the top N entries

Reply:
Okay so I thought I tested this before but wanted double check and I believe it already acts how you want it to. lbinfo always passes nearby as true in order to differentiate the GetLeaderboardData data but if a user is not found (either due to not having an entry or no `u` parameter passed in) then it skips the nearby offset formula and just uses the `o` offset value.

Quick example:
`dorequest.php?r=lbinfo&i=4361&o=25&c=3`
{"User":"Agrahnax","Score":4260,"DateSubmitted":1551543509,"Rank":24,"UserIndex":26},{"User":"ChronoGear","Score":4240,"DateSubmitted":1578425610,"Rank":27,"UserIndex":27},{"User":"MrOwnership","Score":4230,"DateSubmitted":1551562720,"Rank":28,"UserIndex":28}]

`dorequest.php?r=lbinfo&i=4361&u=Searo&o=25&c=3`
{"User":"ssbmLeM","Score":3520,"DateSubmitted":1571791993,"Rank":80,"UserIndex":80},{"User":"Searo","Score":3500,"DateSubmitted":1551566974,"Rank":81,"UserIndex":81},{"User":"Minibits","Score":3500,"DateSubmitted":1558817286,"Rank":81,"UserIndex":82}

If this isn't the expected behavior I can make further modifications

Reply:
Yep. I missed the fact that a failed user lookup for (no user) would result in doing the appropriate thing. Adding a quick check for no user would be a good optimization, but a comment explaining the behavior would also be useful.

> In nearby mode, if a user is specified, adjust the offset to highlight the user. If not in nearby mode, no user was provided, or the user does not have an entry, use the provided offset.